src/runtimes/node/utils/zip.ts

```diff
@@ -291,7 +317,10 @@ const ensurePackageJSONWithModuleType = async function ({
   const normalizedPath = normalizeFilePath({ commonPrefix: basePath, path, userNamespace })
   const contents = JSON.stringify({ type: 'module' })
 
-  addZipContent(archive, contents, normalizedPath)
+  return {
```

Comment:
I've changed this function such that instead of adding the file to the ZIP, it simply returns its content and path. That way, the callers (`createDirectory` and `createZipArchive`) have the power to do what they want with the file.